[PyTorch] Add Device_test.cpp

dhruvbird opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63204
* __->__ #63203

Currently, `c10::Device` isn't being tested - i.e. there's no test to ensure that the device string parsing works as expected. This diff adds very basic tests to assert that the stuff we expect to work works, and the stuff that we don't expect to work doesn't work.

#accept2ship

Differential Revision: [D30286910](https://our.internmc.facebook.com/intern/diff/D30286910/)

dhruvbird committed:
[PyTorch] Add Device_test.cpp

Currently, `c10::Device` isn't being tested - i.e. there's no test to ensure that the device string parsing works as expected. This diff adds very basic tests to assert that the stuff we expect to work works, and the stuff that we don't expect to work doesn't work.

#accept2ship

Differential Revision: [D30286910](https://our.internmc.facebook.com/intern/diff/D30286910/)

[ghstack-poisoned]

dhruvbird committed:
Update on "[PyTorch] Add Device_test.cpp"


Currently, `c10::Device` isn't being tested - i.e. there's no test to ensure that the device string parsing works as expected. This diff adds very basic tests to assert that the stuff we expect to work works, and the stuff that we don't expect to work doesn't work.

#accept2ship

Differential Revision: [D30286910](https://our.internmc.facebook.com/intern/diff/D30286910/)

[ghstack-poisoned]

dhruvbird committed:
Update on "[PyTorch] Add Device_test.cpp"


Currently, `c10::Device` isn't being tested - i.e. there's no test to ensure that the device string parsing works as expected. This diff adds very basic tests to assert that the stuff we expect to work works, and the stuff that we don't expect to work doesn't work.

#accept2ship

Differential Revision: [D30286910](https://our.internmc.facebook.com/intern/diff/D30286910/)

[ghstack-poisoned]

dhruvbird committed:
Update on "[PyTorch] Add Device_test.cpp"


Currently, `c10::Device` isn't being tested - i.e. there's no test to ensure that the device string parsing works as expected. This diff adds very basic tests to assert that the stuff we expect to work works, and the stuff that we don't expect to work doesn't work.

#accept2ship

Differential Revision: [D30286910](https://our.internmc.facebook.com/intern/diff/D30286910/)

[ghstack-poisoned]

raziel reviewed (approved):
Thanks!

There's just the lingering clang issue.

dhruvbird committed:
Update on "[PyTorch] Add Device_test.cpp"


Currently, `c10::Device` isn't being tested - i.e. there's no test to ensure that the device string parsing works as expected. This diff adds very basic tests to assert that the stuff we expect to work works, and the stuff that we don't expect to work doesn't work.

#accept2ship

Differential Revision: [D30286910](https://our.internmc.facebook.com/intern/diff/D30286910/)

[ghstack-poisoned]

dhruvbird committed:
Update on "[PyTorch] Add Device_test.cpp"


Currently, `c10::Device` isn't being tested - i.e. there's no test to ensure that the device string parsing works as expected. This diff adds very basic tests to assert that the stuff we expect to work works, and the stuff that we don't expect to work doesn't work.

#accept2ship

Differential Revision: [D30286910](https://our.internmc.facebook.com/intern/diff/D30286910/)

[ghstack-poisoned]
